### PR TITLE
Feat/crud service

### PR DESCRIPTION
**Pull request by FAEZ10**
# Pull Request Template

## Type of change

- [x] Feature
- [ ] Fix

## Description

This pull request includes the creation of a CRUD service for managing services within the application. The changes include:

- Implementing a new `ServiceController` to handle CRUD operations for services.
- Updating the `Service` entity to include a `uid` field.
- Adding a new `ServiceType` form for service creation and editing.
- Implementing repository methods for service management.
- Creating Twig templates for listing, showing, creating, editing, and deleting services.
- Updating routing to use `uid` instead of `id`.

### Summary of Changes

1. **ServiceController.php**:
   - Created methods for `index`, `new`, `show`, `edit`, `delete`, and `exportAllServices`.
   - Implemented route annotations with `uid`.

2. **Service.php**:
   - Added `uid` field.
   - Added necessary getters and setters.

3. **ServiceRepository.php**:
   - Implemented repository methods for paginating services, counting total services, and counting services by status.

4. **ServiceType.php**:
   - Created a form for handling service creation and updates.

5. **Twig Templates**:
   - Created templates for listing (`index.html.twig`), showing (`show.html.twig`), creating (`new.html.twig`), and editing (`edit.html.twig`) services.
   - Implemented a reusable delete form template (`_delete_form.html.twig`).
   - Updated the table component template (`table.html.twig`) for better UX.

### Dependencies

- Updated to Symfony version `>=6.4`
- PHP version `>=8.3.2`

## How Has This Been Tested?

The following tests were performed to verify the changes:

- **Test A**: Manually tested CRUD operations via the web interface.
- **Test B**: Verified the correct routing and functionality using PHPUnit tests.

**Test Configuration**:

- Symfony version: 6.4
- PHP version: 8.3.2

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: src/Form/ServiceType.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+class ServiceType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('name', TextType::class, [
+                'constraints' => [
+                    new NotBlank(['message' => 'Veuillez saisir un nom valide.']),
+                    new Length(['min' => 3]),
```
Comment (johnnyhelloworld) [inline src/Form/ServiceType.php:29]: il manque EscapeCharacter() pour les caractères spéciaux

File: src/Form/ServiceType.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                'trim' => true,
+                'required' => false,
+            ])
+            ->add('serviceStatus', EntityType::class, [
+                'class' => ServiceStatus::class,
+                'choice_label' => 'name',
+                'placeholder' => 'Choisir un statut',
+            ]);
+    }
+
```
Comment (johnnyhelloworld) [inline src/Form/ServiceType.php:77]: il manque ucfirst() et/ou strtolower() pour bien formatter les données

File: templates/components/Table.html.twig [hunk 3]
```diff
@@ -1,72 +1,61 @@
...
                 </div>
-            {% endif %}
+            {% endfor %}
         </div>
     </div>
-    
-    {% if rows|length == 0 %}
-        <div class="w-full py-10 text-center">
-            Pas de données.
+    {% if pagination is defined %}
```
Comment (johnnyhelloworld) [inline templates/components/Table.html.twig]: on a pas encore mergé ma branche mais c'est paginatedData et non pagination

File: templates/dashboard/service/show.html.twig [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+<title>Service</title>
+
+{% block body %}
+    <h1>Service</h1>
+
+    <table class="table">
+        <tbody>
+            <tr>
+                <th>Id</th>
```
Comment (johnnyhelloworld) [inline templates/dashboard/service/show.html.twig]: remplace l'id par l'uuid pour plus de sécurité

File: templates/dashboard/service/show.html.twig [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
+
+<title>Service</title>
```
Comment (johnnyhelloworld) [inline templates/dashboard/service/show.html.twig]: traduction sur cette page

File: src/Repository/ServiceRepository.php [hunk 6]
```diff
@@ -34,4 +34,34 @@ public function paginateServicesByCompany(User $user, int $page): PaginationInte
             5
         );
     }
+
+    public function countTotalServices(): int
```
Comment (johnnyhelloworld) [inline src/Repository/ServiceRepository.php]: c'est countTotalServicesByCompany() -> à confirmer si on affiche tous les services ou seulement nos services à nous

Comment (FAEZ10) [reply src/Repository/ServiceRepository.php]: C'est logique que t'as accès uniquement  aux services de ton entreprise

File: templates/dashboard/service/show.html.twig (hunk 5 above)
Comment (FAEZ10) [reply templates/dashboard/service/show.html.twig]: Noté

File: templates/dashboard/service/show.html.twig (hunk 4 above)
Comment (FAEZ10) [reply templates/dashboard/service/show.html.twig]: Noté

File: src/Repository/ServiceRepository.php (hunk 6 above)
Comment (johnnyhelloworld) [reply src/Repository/ServiceRepository.php]: Ok, du coup c'est countTotalServicesByCompany()

**Pull request merged**